x/concentrated-liquidity/math/tick.go

```diff
@@ -120,55 +120,59 @@ func PriceToTick(price sdk.Dec) (sdk.Int, error) {
 
 // PriceToTickRoundDown takes a price and returns the corresponding tick index.
 // If tickSpacing is provided, the tick index will be rounded down to the nearest multiple of tickSpacing.
-func PriceToTickRoundDown(price sdk.Dec, tickSpacing uint64) (sdk.Int, error) {
+func PriceToTickRoundDown(price sdk.Dec, tickSpacing uint64) (int64, error) {
 	tickIndex, err := PriceToTick(price)
 	if err != nil {
-		return sdk.Int{}, err
+		return 0, err
 	}
 
 	// Round the tick index down to the nearest tick spacing if the tickIndex is in between authorized tick values
-	tickSpacingInt := sdk.NewIntFromUint64(tickSpacing)
-	tickIndexModulus := tickIndex.Mod(tickSpacingInt)
-	if !tickIndexModulus.Equal(sdk.ZeroInt()) {
-		tickIndex = tickIndex.Sub(tickIndexModulus)
+	// Note that this is Euclidean modulus.
```

Comment:
whats a euclidean modulus

Reply:
https://en.wikipedia.org/wiki/Euclidean_algorithm ?

is there a relevant detail about that algo here?

Reply:
If not lets delete this.

(As someone whose implemented modular reduction many times, I have no idea what meaning is being conveyed to me with this comment)

Reply:
If its that, when `tickIndex is negative, we get `tickspacing - (|tickInex| % tickspacing)`, then we can add a comment that thats what were dealing with here!

Reply:
So the Go's default modulus and Euclidean modules differ for negative divisors.

The Go's default would not work. As a result, we do:
```go
tickIndexModulus := tickIndex % int64(tickSpacing)
if tickIndexModulus < 0 {
  tickIndexModulus += int64(tickSpacing)
}
```

This is the algo:
```go
func euclideanModulus(dividend, divisor int) int {
    modulus := dividend % divisor
    if modulus < 0 {
        modulus += divisor
    }
    return modulus
}
```
Example rounding down logic
```
tickIndex = -17
tickSpacing = 10
tickIndexModulus = 17 % 10 = 3
tickIndex = -17 -3 = -20
```

I will break this down more in the comment. I think it didn't do a good job of explaining this